src/core/battle/state/apply.rs

```diff
@@ -339,9 +339,13 @@ fn apply_effect_wound(state: &mut State, id: Id, effect: &effect::Wound) {
 fn apply_effect_knockback(state: &mut State, id: Id, effect: &effect::Knockback) {
     assert!(state.map().is_inboard(effect.from));
     assert!(state.map().is_inboard(effect.to));
+
     assert!(!state::is_tile_blocked(state, effect.to));
     let parts = state.parts_mut();
-    parts.pos.get_mut(id).0 = effect.to;
+    if effect.strength.can_push(parts.blocker.get(id).weight) {
+        parts.pos.get_mut(id).0 = effect.to;
+    }
+    // else show a resistance ?
```

Comment:
I havn't looked into graphics yet, I guess resistance feedback will be in another pull request

Reply:
Thanks, I implemented it